utils/staking-miner/src/monitor.rs

```diff
@@ -131,39 +205,57 @@ macro_rules! monitor_cmd_for { ($runtime:tt) => { paste::paste! {
 			config: MonitorConfig,
 		) {
 
+			async fn flatten<T>(
+				handle: tokio::task::JoinHandle<Result<T, StakingMinerError>>
+			) -> Result<T, StakingMinerError> {
+				match handle.await {
+					Ok(Ok(result)) => Ok(result),
+					Ok(Err(err)) => Err(err),
+					Err(err) => panic!("tokio spawn task failed; kill task: {:?}", err),
+				}
+			}
+
 			let hash = at.hash();
 			log::trace!(target: LOG_TARGET, "new event at #{:?} ({:?})", at.number, hash);
 
-			// if the runtime version has changed, terminate.
-			if let Err(err) = crate::check_versions::<Runtime>(&rpc).await {
-				let _ = tx.send(err.into());
-				return;
-			}
-
-			// we prefer doing this check before fetching anything into a remote-ext.
-			if ensure_signed_phase::<Runtime, Block>(&rpc, hash).await.is_err() {
-				log::debug!(target: LOG_TARGET, "phase closed, not interested in this block at all.");
-				return;
-			}
+			let rpc1 = rpc.clone();
+			let rpc2 = rpc.clone();
+			let rpc3 = rpc.clone();
+			let rpc4 = rpc.clone();
+			let account = signer.account.clone();
+
+			let signed_phase_fut = tokio::spawn(async move {
+				ensure_signed_phase::<Runtime, Block>(&rpc1, hash).await
+			});
+
+			let no_prev_sol_fut = tokio::spawn(async move {
+				ensure_no_previous_solution::<Runtime, Block>(&rpc2, hash, &account).await
+			});
+
+			let check_version_fut = tokio::spawn(async move {
+				crate::check_versions::<Runtime>(&rpc3).await
+			});
+
+			let ext_fut = tokio::spawn(async move {
+				crate::create_election_ext::<Runtime, Block>(rpc4, Some(hash), vec![]).await
+			});
+
+			// Run the calls concurrently and return once all has completed or any failed.
+			let res = tokio::try_join!(
```

Comment:
//cc @emostov now this these calls are performed in parallel however the code is a bit nasty with the clones

Reply:
I'm going to revert the `create_election_ext` because it increases network usage significantly even if the we drop the requests the server will answer it anyway.

In addition we only need to fetch these in signed phase anyway still no performance issues what I'm aware of anyway.